### ISSUE TITLE
`Style/MethodCallWithoutArgsParentheses` safe autocorrection produces invalid syntax when the parentheses contain a comment

### ISSUE THREAD

**Issue by KirtashW17** [rubocop/rubocop#15426]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

Incorrect autocorrect

### Cop name

Style/MethodCallWithoutArgsParentheses

### Steps to reproduce the problem

1. Create a file `repro.rb`:

   ```ruby
   x = Foo.new(
     # just a comment
   ) do |c|
     c.bar
   end
   ```

2. Run the cop's safe autocorrection:

   ```bash
   rubocop -a --only Style/MethodCallWithoutArgsParentheses repro.rb
   ```

### Expected behavior

Either the parentheses are left alone (they are not empty, they contain a comment
and the call is a block receiver), or the comment is preserved in a position that
keeps the code parseable. The corrected file must remain valid Ruby.

### Actual behavior

The parentheses are removed, stranding the comment and orphaning the `do?end` block.
The result no longer parses:

```ruby
x = Foo.new
  # just a comment
 do |c|
  c.bar
end
```

### RuboCop configuration

```yaml

```

### RuboCop version

```shell
1.88.1
```
